### PR TITLE
AVX-59376 Add a E2E test

### PR DESCRIPTION
**Pull request by avx-rchung**
This test exercises the full Controller HA lifecycle

### CODE REVIEW

**Comment by avx-rchung** [comment]
One more, thanks for all the reviews :) I verified this e2e test works but it needs a few more fixes for the controller that I need to submit separately

File: tests/terraform/e2e_controller/main.tf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  owners      = ["600530653622"]
+
+  filter {
+    name   = "name"
+    values = ["avx-controller-g4-*"]
+  }
+}
+
+# try to satisfy the controller's password strength meter by combining
+# two passwords with a special char
```
Comment (dnardo) [inline tests/terraform/e2e_controller/main.tf:12]: Can this fail sometimes to meet that requirement?  The docstring makes it it sound like maybe?

File: tests/test_e2e.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+from aviatrix_ha.api import client
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+@terraform("e2e_controller", scope="session")
+@pytest.mark.skipif(os.environ.get("TF_VAR_customer_id") is None, reason="TF_VAR_customer_id is not set")
+def test_e2e_controller(e2e_controller):
```
Comment (dnardo) [inline tests/test_e2e.py:37]: nice test.

File: tests/terraform/e2e_controller/main.tf [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  controller_admin_email    = var.admin_email
+  controller_admin_password = local.admin_password
+  incoming_ssl_cidrs        = var.incoming_ssl_cidrs
+
+  controller_ami_id = var.controller_ami_id == "" ? data.aws_ami.latest_controller.id : var.controller_ami_id
+  controller_user_data = templatefile("${path.module}/cloud-config.yaml", {
+    software_version          = var.controller_version
+    use_containerized_gateway = var.use_containerized_gateway ? "true" : "false"
+  })
+  controller_wait_for_setup_duration = "0s"
```
Comment (dnardo) [inline tests/terraform/e2e_controller/main.tf:39]: Curious what this does and why we set it to 0s ?  I couldn't find in the referenced module.

**Review by dnardo — APPROVED** [review]
Looks great, thanks!

File: tests/terraform/e2e_controller/main.tf (hunk 1 above)
Comment (avx-rchung) [reply tests/terraform/e2e_controller/main.tf:12]: I have not seen it fail. This was initially written by Ross/Joe for their testing and has gone through a couple of iterations, and it seems to work well. I can remove the comment if it's confusing

File: tests/test_e2e.py (hunk 2 above)
Comment (avx-rchung) [reply tests/test_e2e.py:37]: thanks! i got sick of clicking through many different UIs to do the test...

File: tests/terraform/e2e_controller/main.tf (hunk 3 above)
Comment (avx-rchung) [reply tests/terraform/e2e_controller/main.tf:39]: the controller-init module is common for pre-containerized and containerized controllers. In the pre-containerized case, the "initial_setup" API finishes after phase 1 (download software) but we still need some time for phase 2 to complete, so it sleeps for 10 minutes (!) before continuing. In the containerized case we don't need this 10 minute sleep.

Comment (dnardo) [reply tests/terraform/e2e_controller/main.tf:39]: Nice

**Pull request merged**